### PR TITLE
adding multi-modal pre-processor for cohere

### PR DESCRIPTION
**Pull request by dhrubo-os**
### Description
[adding multi-modal pre-processor for cohere

Cohere multi-modal model either accepts array of texts or array of base64 images (but only 1 item in the array). So if customer wants to use images then then they can use this pre-processor. 
]

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/ml/common/connector/functions/preprocess/CohereMultiModalEmbeddingPreProcessFunction.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if (docs.isEmpty() || (docs.size() == 1 && docs.getFirst() == null)) {
+            throw new IllegalArgumentException("No image provided");
+        }
+    }
+
+    @Override
+    public RemoteInferenceInputDataSet process(MLInput mlInput) {
+        TextDocsInputDataSet inputData = (TextDocsInputDataSet) mlInput.getInputDataset();
+        Map<String, String> parametersMap = new HashMap<>();
+        parametersMap.put("images", inputData.getDocs().getFirst());
```
Comment (ylwu-amzn) [inline common/src/main/java/org/opensearch/ml/common/connector/functions/preprocess/CohereMultiModalEmbeddingPreProcessFunction.java:37]: Only support image input? Should we also consider support text input ?

Comment (dhrubo-os) [reply common/src/main/java/org/opensearch/ml/common/connector/functions/preprocess/CohereMultiModalEmbeddingPreProcessFunction.java:37]: Cohere multi-modal doesn't support text with image input. Either image input or text input. [Example notebook](https://github.com/cohere-ai/notebooks/blob/main/notebooks/Multimodal_Semantic_Search.ipynb)

For text input, we could use our regular one: `connector.pre_process.cohere.embedding`

Comment (dhrubo-os) [reply common/src/main/java/org/opensearch/ml/common/connector/functions/preprocess/CohereMultiModalEmbeddingPreProcessFunction.java:37]: Just one per request.

Comment (dhrubo-os) [reply common/src/main/java/org/opensearch/ml/common/connector/functions/preprocess/CohereMultiModalEmbeddingPreProcessFunction.java:37]: An array of image data URIs for the model to embed. Maximum number of images per call is 1.

Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/connector/functions/preprocess/CohereMultiModalEmbeddingPreProcessFunction.java:37]: Got it, suggest add more java doc to explain these details

**Review by ylwu-amzn — APPROVED** [review]

**Comment by R1V3RJ1s** [comment]
Regarding to this PR, do we need to do any alternations on current blueprint?

**Comment by b4sjoo** [comment]
Regarding to this PR, do we need to do any alternations on current blueprint?

**Comment by dhrubo-os** [comment]
> Regarding to this PR, do we need to do any alternations on current blueprint?

This is a new pre-process function, so I'll add a new blue print doc.

**Review by b4sjoo — APPROVED** [review]

**Pull request merged**